### DISCUSSION TITLE
Using Fog computing on CloudSimPLus

### DISCUSSION THREAD

**Discussion by Nadjib23** [manoelcampos/cloudsim-plus#402] [Q&A] [answered]
Hi,
I find CloudSimPlus really easy to use and more complete than classic CloudSim. I'm currently working on Fog computing using iFogSim, which heavily relies on CloudSim, especially for cloudlet scheduling. However, the implementation of time shared cloudlet scheduler is not realistic contrary to the Completely Fair Linux Scheduler in CloudSimPlus.
Is it possible to use CloudSimPlus with iFogSim or is there a way to adapt it ?

**Comment by manoelcampos** [comment, marked as answer]
Hello @Nadjib23 
I don't think there is any easy way to adapt iFogSim to use CloudSim Plus as the underlying simulation framework.
[PureEdgeSim](https://github.com/CharafeddineMechalikh/PureEdgeSim) may be what you are looking for. It's based on CloudSim Plus, using it as a dependency instead of just naively  copying the framework code and change it.
